### DISCUSSION TITLE
Metrics support? (eg. Grafana, statsd, etc.)

### DISCUSSION THREAD

**Discussion by jjwwiodyne** [fastapi/fastapi#8070] [Questions]
**Description**

Can FastAPI be configured to transmit API usage metrics to a metrics server (e.g. Grafana, statsd, etc.? For example, success and failures for API calls (api.api_call.error.404)

**Comment by dmontagu** [comment]
This would typically be done via an ASGI middleware.

Here is an example that may work for you out of the box, or that you may use as a starting point for your own implementation: https://github.com/steinnes/timing-asgi

**Comment by jjwwiodyne** [comment]
Perfect. Thank you!

**Comment by tiangolo** [comment]
Thanks for the help here @dmontagu ! :clap: :bow: 

Thanks for reporting back and closing the issue @jjwwiodyne  :+1:
